src/bindings/python/flux/accounting/job_archive_interface.py

```diff
@@ -288,6 +311,77 @@ def calc_usage_factor(
 
     usage_user_current = sum(per_job_factors)
 
+    # if no new jobs were found and we are still in the same half-life
+    # period, then the current usage remains the same
+    if len(job_totals_user) == 0 and (
+        float(t_end_half_life_period) > (time.time() - half_life_period)
```

Comment:
I think this logic is perfectly fine.  Typically, using `time.time()` directly can cause issues with testing.  Sidenote (just learned this today), we can use `unittest`s `mock` to override `time.time` within our tests: https://stackoverflow.com/questions/35783760/faking-a-time-stamp-from-time-time-in-nose